de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/NegotiationFactory.java

```diff
@@ -129,14 +131,12 @@ public IncomingSessionNegotiation newIncomingSessionNegotiation(
 
   public AbstractOutgoingProjectNegotiation newOutgoingProjectNegotiation(
       final JID remoteAddress,
-      final TransferType transferType,
       final List<IProject> resources,
       final ISarosSessionManager sessionManager,
       final ISarosSession session) {
 
-    if (transferType == null) {
-      throw new IllegalArgumentException("transferType must not be null");
-    }
+    User remoteUser = session.getUser(remoteAddress);
```

Comment:
Duplicated code. Why not use getTransferType(ISarosSession session, JID remoteAddress) as signature ? Also the user may have already left the session and so this call would return null.

Reply:
Didn't removed the duplicate because I plan to change the signature of `new[In|Out]comingProjectNegotiation` to use `User`, but yes better change it for now.